src/application/redux/reducers/wallet-reducer.ts

```diff
@@ -28,13 +28,28 @@ export const walletInitState: WalletState = {
   },
   updaterLoaders: 0,
   isVerified: false,
+  lockedUtxos: {},
+};
+
+// returns only utxos locked for less than 5 minutes
+const filterOnlyRecentLockedUtxos = (state: WalletState) => {
```

Comment:
I think u can simplify and optimize this using a redux-saga (tdex-app [example](https://github.com/tdex-network/tdex-app/blob/ce03725b0d4a82f6d0092caa1eeea3baa4fa9520/src/redux/sagas/walletSaga.ts#L106-L110)) instead of a filter function. Here, it looks like the LockedUtxos are only "unlocked" if an unspent is added (where the filter function is called).

Reply:
Agree with @bordalix

Reply:
Shall we move this in a redux saga?

> it looks like the LockedUtxos are only "unlocked" if an unspent is added
@louisinger  what would be the problem of not unlocking? As soon a transaction is submitted to mempool, can't be canceled (unless replace by fee, but at this point the new utxo state will be synced from network)

Reply:
we agreed offline with @bordalix to address in this PR only the Louis concern, but not about the no-network issue (can be done later on in a dedicated PR with proper network handling now is missing)